types/plugin.d.ts

```diff
@@ -1,6 +1,12 @@
+/* eslint-disable @typescript-eslint/ban-types */
+
 import { FastifyInstance } from './instance'
 import { RawServerBase, RawRequestDefaultExpression, RawReplyDefaultExpression, RawServerDefault } from './utils'
 
+export interface FastifyPluginOptions {
+  [key:string]: any
+}
```

Comment:
```suggestion
export type FastifyPluginOptions = Record<string, any>
```

Reply:
had the issue with the empty interface but yes it works for type indeed!